File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        Map<Integer, Integer> remainders = new HashMap<>();

        for(int i = 0; i < nums.length; i++){
            if(remainders.containsKey(target - nums[i])){
                return new int[]{i, remainders.get(target - nums[i])};
            }
            remainders.put(nums[i], i);
        }
        return new int[]{};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        num_dicts = {}

        for i in range(len(nums)):
            remainder = target - nums[i]
            if remainder in num_dicts:
                return [i, num_dicts[remainder]]
            num_dicts[nums[i]] = i